packages/legend-graph/src/models/DomainModifierHelper.ts

```diff
@@ -0,0 +1,373 @@
+ * limitations under the License.
+ */
+
+import { addUniqueEntry, changeEntry, deleteEntry } from '@finos/legend-shared';
+import { action } from 'mobx';
+import type { AbstractProperty } from './metamodels/pure/packageableElements/domain/AbstractProperty';
+import type { AnnotatedElement } from './metamodels/pure/packageableElements/domain/AnnotatedElement';
+import { Association } from './metamodels/pure/packageableElements/domain/Association';
+import type { Class } from './metamodels/pure/packageableElements/domain/Class';
+import type { ConcreteFunctionDefinition } from './metamodels/pure/packageableElements/domain/ConcreteFunctionDefinition';
+import type { Constraint } from './metamodels/pure/packageableElements/domain/Constraint';
+import type { DerivedProperty } from './metamodels/pure/packageableElements/domain/DerivedProperty';
+import type { Enum } from './metamodels/pure/packageableElements/domain/Enum';
+import type { Enumeration } from './metamodels/pure/packageableElements/domain/Enumeration';
+import type { EnumValueReference } from './metamodels/pure/packageableElements/domain/EnumValueReference';
+import type { GenericType } from './metamodels/pure/packageableElements/domain/GenericType';
+import type { GenericTypeReference } from './metamodels/pure/packageableElements/domain/GenericTypeReference';
+import type {
+  Measure,
+  Unit,
+} from './metamodels/pure/packageableElements/domain/Measure';
+import type { Multiplicity } from './metamodels/pure/packageableElements/domain/Multiplicity';
+import type { Package } from './metamodels/pure/packageableElements/domain/Package';
+import type { Profile } from './metamodels/pure/packageableElements/domain/Profile';
+import type { Property } from './metamodels/pure/packageableElements/domain/Property';
+import type { PropertyReference } from './metamodels/pure/packageableElements/domain/PropertyReference';
+import type { Stereotype } from './metamodels/pure/packageableElements/domain/Stereotype';
+import type { StereotypeReference } from './metamodels/pure/packageableElements/domain/StereotypeReference';
+import type { Tag } from './metamodels/pure/packageableElements/domain/Tag';
+import type { TaggedValue } from './metamodels/pure/packageableElements/domain/TaggedValue';
+import type { TagReference } from './metamodels/pure/packageableElements/domain/TagReference';
+import type { Type } from './metamodels/pure/packageableElements/domain/Type';
+import type { PackageableElement } from './metamodels/pure/packageableElements/PackageableElement';
+import type { PackageableElementReference } from './metamodels/pure/packageableElements/PackageableElementReference';
+import type { RawLambda } from './metamodels/pure/rawValueSpecification/RawLambda';
+import type { RawVariableExpression } from './metamodels/pure/rawValueSpecification/RawVariableExpression';
+
+// PackageableElementReference
+export const setPackageableElementReferenceValue = action(
```

Comment:
I think places in the metamodels that we're still using this method, we could probably to swap out to just do `ref.value = value`.